webview/src/plots/components/customPlots/util.ts

```diff
@@ -1,6 +1,99 @@
 import { VisualizationSpec } from 'react-vega'
+import { ColorScale } from 'dvc/src/plots/webview/contract'
 
-export const createSpec = (metric: string, param: string) =>
+export const createCheckpointSpec = (
+  title: string,
+  param: string,
+  scale?: ColorScale
+): VisualizationSpec =>
+  ({
+    $schema: 'https://vega.github.io/schema/vega-lite/v5.json',
+    data: { name: 'values' },
+    encoding: {
+      color: {
+        field: 'group',
+        legend: { disable: true },
+        scale,
+        title: 'rev',
+        type: 'nominal'
+      },
+      x: {
+        axis: { format: '0d', tickMinStep: 1 },
+        field: 'iteration',
+        title: param,
+        type: 'quantitative'
+      },
+      y: {
+        field: 'y',
+        scale: { zero: false },
+        title,
+        type: 'quantitative'
+      }
+    },
+    height: 'container',
+    layer: [
+      {
+        layer: [
+          { mark: { type: 'line' } },
+          {
+            mark: { type: 'point' },
+            transform: [
+              {
+                filter: { empty: false, param: 'hover' }
+              }
+            ]
+          }
+        ]
+      },
+      {
+        encoding: {
+          opacity: { value: 0 },
+          tooltip: [
+            { field: 'group', title: 'name' },
+            {
+              field: 'y',
+              title: title.slice(Math.max(0, title.indexOf(':') + 1)),
```

Comment:
Shouldn't this be done before `truncateVerticalTitle` (on the extension side)?

Reply:
Sounds like a good idea! Will add this to our followup checklist :)

Reply:
I feel like breaking changes introduces in this PR should be fixed in this PR, unless it's something really big or introduces a new concern.

Reply:
> I feel like breaking changes introduces in this PR should be fixed in this PR, unless it's something really big or introduces a new concern.

Of course! I technically didn't add this line of code so I didn't realize it was a breaking change :)